pkg/capacity/manager_test.go

```diff
@@ -8,323 +8,171 @@ import (
-		InitializeCounters(t, toolchainStatus)
-
-		// when
-		clusterName, err := capacity.NewClusterManager(commontest.HostOperatorNs, fakeClient).GetOptimalTargetCluster(
-			ctx,
-			capacity.OptimalTargetClusterFilter{
-				PreferredCluster:         "member1",
-				ToolchainStatusNamespace: commontest.HostOperatorNs,
-			},
-		)
-
-		// then
-		require.NoError(t, err)
-		assert.Equal(t, "member2", clusterName)
-	})
-
-	t.Run("with two clusters, none of them is returned since it reaches max number of Spaces, no matter what is defined as preferred", func(t *testing.T) {
+	t.Run("with one cluster and enough capacity", func(t *testing.T) {
 		// given
-		spc1 := hspc.NewEnabledValidTenantSPC("member1", spc.MaxNumberOfSpaces(1), spc.MaxMemoryUtilizationPercent(60))
-		spc2 := hspc.NewEnabledValidTenantSPC("member2", spc.MaxNumberOfSpaces(1), spc.MaxMemoryUtilizationPercent(95))
-		fakeClient := commontest.NewFakeClient(t, toolchainStatus, spc1, spc2)
-		InitializeCounters(t, toolchainStatus)
+		spaceProvisionerConfig := hspc.NewEnabledValidTenantSPC("member1")
+		status := test.NewToolchainStatus()
+		fakeClient := commontest.NewFakeClient(t, spaceProvisionerConfig, status)
+		test.InitializeCounters(t, status)
+		cm := capacity.NewClusterManager(commontest.HostOperatorNs, runtimeclient.Client(fakeClient))
 
 		// when
-		clusterName, err := capacity.NewClusterManager(commontest.HostOperatorNs, fakeClient).GetOptimalTargetCluster(
-			ctx,
-			capacity.OptimalTargetClusterFilter{
-				PreferredCluster:         "member2",
-				ToolchainStatusNamespace: commontest.HostOperatorNs,
-			},
-		)
+		clusterName, err := cm.GetOptimalTargetCluster(ctx, capacity.OptimalTargetClusterFilter{})
 
 		// then
 		require.NoError(t, err)
-		assert.Equal(t, "", clusterName)
+		assert.Equal(t, "member1", clusterName)
 	})
 
-	t.Run("with two clusters but only the second one has enough capacity - using the default values", func(t *testing.T) {
+	t.Run("with three eligible clusters it returns the one with the most capacity (the first one)", func(t *testing.T) {
 		// given
-		spc1 := hspc.NewEnabledValidTenantSPC("member1", spc.MaxMemoryUtilizationPercent(62))
-		spc2 := hspc.NewEnabledValidTenantSPC("member2", spc.MaxMemoryUtilizationPercent(62))
-		fakeClient := commontest.NewFakeClient(t, toolchainStatus, spc1, spc2)
-		InitializeCounters(t, toolchainStatus)
-
-		// when
-		clusterName, err := capacity.NewClusterManager(commontest.HostOperatorNs, fakeClient).GetOptimalTargetCluster(
-			ctx,
-			capacity.OptimalTargetClusterFilter{
-				ToolchainStatusNamespace: commontest.HostOperatorNs,
-			},
-		)
+		spc1 := hspc.NewEnabledValidTenantSPC("member1", spc.MaxNumberOfSpaces(10000), spc.WithConsumedSpaceCount(700))
+		spc2 := hspc.NewEnabledValidTenantSPC("member2", spc.MaxNumberOfSpaces(1000), spc.WithConsumedSpaceCount(700))
+		spc3 := hspc.NewEnabledValidTenantSPC("member3", spc.MaxNumberOfSpaces(1000), spc.WithConsumedSpaceCount(200))
```

Comment:
within the context of the capacity manager, we shouldn't care about the consumed capacity set in the SPC status
```suggestion
		spc1 := hspc.NewEnabledValidTenantSPC("member1", spc.MaxNumberOfSpaces(10000))
		spc2 := hspc.NewEnabledValidTenantSPC("member2", spc.MaxNumberOfSpaces(1000))
		spc3 := hspc.NewEnabledValidTenantSPC("member3", spc.MaxNumberOfSpaces(1000))
```